include/sys_clock.h

```diff
@@ -49,6 +49,7 @@ typedef u32_t k_ticks_t;
 #endif
 
 #define K_TICKS_FOREVER ((k_ticks_t) -1)
+#define K_MS_FOREVER ((k_ticks_t) -1)
```

Comment:
I wonder if `k_ticks_t` type is the (semantically) correct type for this; shouldn't this be just an integer value?

Reply:
IMO, yes.  `k_ticks_t` is the integral type used for tick counts.  It's likely to be compared against `k_ticks_t` values so it should have the same signedness too, hence the cast.